app/src/main/java/com/example/mona/facebookoffline/FacebookApiService.java

```diff
@@ -0,0 +1,91 @@
+package com.example.mona.facebookoffline;
+
+import android.os.Bundle;
+import android.util.Log;
+
+import com.facebook.AccessToken;
+import com.facebook.GraphRequest;
+import com.facebook.GraphResponse;
+import com.facebook.HttpMethod;
+
+import org.json.JSONException;
+
+/**
+ * Singleton class that handles requests to Facebook's Graph Api
+ *
+ * Created by mona on 4/29/16.
+ */
+public class FacebookApiService {
+    private static final String TAG = FacebookApiService.class.getSimpleName();
+
+    /** Constants used as parameters/fields for the Fb Graph Api */
+    private static final String FIELDS = "fields";
+    private static final String ACCESS_TOKEN = "access_token";
+    private static final String MESSAGE = "message";
+
+    public void postMessageToPage(final String pageId, final String message,
+                                  final GraphRequest.Callback cb) {
+        Bundle params = new Bundle();
+        params.putString(FIELDS, ACCESS_TOKEN);
+
+        // TODO(mona): Currently we're requesting a page token every time, which is probably not necessary
+        new GraphRequest(
+                AccessToken.getCurrentAccessToken(),
+                "/" + Constants.PAGE_ID,
+                params,
+                HttpMethod.GET,
+                new GraphRequest.Callback() {
+
+                    @Override
+                    public void onCompleted(GraphResponse response) {
+
+                        if (response.getError() != null) {
+                            Log.w(TAG, "Error fetching page access token", response.getError().getException());
+                            notifyError(cb, response);
+                        } else {
+                            try {
+                                String pageToken = response.getJSONObject().getString(ACCESS_TOKEN);
+                                Log.d(TAG, "pageToken=" + pageToken);
+
+                                executeRequestToPost(pageId, pageToken, message, cb);
+                            } catch (JSONException e) {
+                                Log.w(TAG, "Unable to properly parse response: " + response, e);
+                                notifyError(cb, response);
+                            }
+                        }
+
+                    }
+                }
+
+        ).executeAsync();
+    }
+
+    private void notifyError(GraphRequest.Callback cb, GraphResponse response) {
+        cb.onCompleted(response);
+    }
+
+    private void executeRequestToPost(String pageId, String pageAccessToken, String message,
+                                  GraphRequest.Callback cb) {
+
+        Bundle params = new Bundle();
+        params.putString(MESSAGE, message);
+
+        AccessToken pageToken = createAccessToken(pageAccessToken, AccessToken.getCurrentAccessToken());
```

Comment:
Actually, I think I understand. On line 47 you pull the string `pageAccessToken` out of the JSON response. Is it possible to pull that out of the `GraphResponse` object in the form of an `AccessToken`? That would allow us to avoid this `createAccessToken` method. I haven't read the docs, I'm just speculating here.